jwt/jwt_test.go

```diff
@@ -574,27 +584,41 @@ func TestNewValidator(t *testing.T) {
 		{
 			name: "new validator with keySet",
 			args: args{
-				keySet: func() KeySet {
+				keySets: func() []KeySet {
 					ks, err := NewJSONWebKeySet(context.Background(),
 						"https://issuer.com/"+wellKnownJWKS, "")
 					require.NoError(t, err)
-					return ks
+					return []KeySet{ks}
 				},
 			},
 		},
 		{
 			name: "new validator with nil keySet",
 			args: args{
-				keySet: func() KeySet {
+				keySets: func() []KeySet {
 					return nil
 				},
 			},
 			wantErr: true,
 		},
+		{
```

Comment:
Could add one more case for the nil key set in the list of key sets. Thanks for adding that validation!

```
{
	name: "new validator with nil keySet in keySets",
	args: args{
		keySets: func() []KeySet {
			ks, err := NewJSONWebKeySet(context.Background(),
				"https://issuer.com/"+wellKnownJWKS, "")
			require.NoError(t, err)
			return []KeySet{ks, nil}
		},
	},
	wantErr: true,
},
```